homeassistant/components/bond/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Bond integration."""
+import logging
+
+from bond import Bond
+from requests.exceptions import ConnectionError as RequestConnectionError
+from simplejson import JSONDecodeError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_ACCESS_TOKEN): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    def authenticate(bond_hub: Bond) -> bool:
+        try:
+            bond_hub.getDeviceIds()
+            return True
+        except RequestConnectionError:
+            raise CannotConnect
+        except JSONDecodeError:
+            return False
+
+    bond = Bond(data[CONF_HOST], data[CONF_ACCESS_TOKEN])
+
+    if not await hass.async_add_executor_job(authenticate, bond):
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": data[CONF_HOST]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Bond."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
It's nice to only have the line that can raise within the try... except. We can move this to an `else:` block.

Reply:
thanks, applied